installer/scripts/service_control

```diff
@@ -302,167 +518,158 @@ disable_omsagent_service()
-        exit $?
-        ;;
-
-    start)
-        if [ -z $2 ]; then
-            start_all_omsagent
-        else
-            setup_variables $2
-            start_omsagent
-        fi
-        ;;
-
-    stop)
-        if [ -z $2 ]; then
-            stop_all_omsagent
-        else
-            setup_variables $2
-            stop_omsagent
+	do
+        if setup_variables $ws_id; then
+            $_cmdArg
         fi
-        ;;
+        fcso_increment $_cmdArg $ws_id $WS_STATUS
+	done
+}
 
-    restart)
-        if [ -z $2 ]; then
-            restart_all_omsagent
-        else
-            setup_variables $2
-            restart_omsagent
-        fi
-        ;;
+service_template() {
+	local _activityName=$1
+	local _wsId=$2
 
-    reload)
-        # TODO: Due to a bug in OMS right now, we can't reload via a signal
-        if [ -z $2 ]; then
-            restart_all_omsagent
-        else
-            setup_variables $2
-            restart_omsagent
+    verify_privileges
+	l_procedure=$(generateProcedureName $_activityName)
+	if [ -n "$_wsId" ]
+	then
+        if setup_variables $_wsId; then
+            $l_procedure
         fi
-        ;;
-
-    find-systemd-dir)
-        find_systemd_dir
-        ;;
+		return $WS_STATUS
+	fi
+
+	fcso_new
+	check_oms_and_invoke $l_procedure
+    check_non_oms_and_invoke $l_procedure
+	WS_STATUS=$FCSO_FailCount
+	return $WS_STATUS
+}
 
-    enable)
-        if [ -z $2 ]; then
-            enable_all_omsagent_services
-        else
-            setup_variables $2
-            enable_omsagent_service
-        fi
-        ;;
+# ## Usage
+
+printUsage() {
+	cat <<EOUSAGE
+	USAGE:  $0 -h
+	USAGE:  $0 help
+	USAGE:  $0 <script command> [workspace id]
+	Where script commands are any of the following:
+		-h or help
+		disable:		 To disable the OMS service, leaving it installed.
+		endable:		 To enable the OMS service after being disabled.
+		find-systemd-dir To establish directory of your OS's systemd, if it has one.
+		is-running:		 Boolean whether or not omsagent is running.  This is a 1
+                         for true, and 0 for false, unlike script booleans (Legacy).
+		reload:  		 Reload agent configuration (at this time same as restart below)
+		restart:		 Restart the OMS service via the service control manager
+		start:  		 Start the OMS service via the service control manager
+		stop:  			 Stop the OMS service via the service control manager
+
+	Note: Bourne Shell / Dash / POSIX apparently does not see arguments in a
+	source evocation.  So to get the WORKSPACE_ID to a sourced copy, unless it
+	is in Bourne shell or another shell that sees $# and $1, $2,..., you need to
+	define the variable beforehand, which this draft sees under the name:
+		WORKSPACE_ID 
+EOUSAGE
+}
 
-    disable)
-        if [ -z $2 ]; then
-	    disable_all_omsagent_services
-        else
-            setup_variables $2
-	    disable_omsagent_service
-        fi
-        ;;
+# ## Main Procedure
 
+case "$1" in
```

Comment:
Please use styling for this case block like bundle_skel uses as well. As it is below, the spacing is inconsistent, and some cases span multiple lines while others do not.